src/topology/category/Top/opens.lean

```diff
@@ -16,23 +35,61 @@ namespace topological_space.opens
 
 variables {X Y Z : Top.{u}}
 
-instance opens_category : category.{u} (opens X) :=
-{ hom  := λ U V, ulift (plift (U ≤ V)),
-  id   := λ X, ⟨ ⟨ le_refl X ⟩ ⟩,
-  comp := λ X Y Z f g, ⟨ ⟨ le_trans f.down.down g.down.down ⟩ ⟩ }
+/-!
+Since `opens X` has a partial order, it automatically receives a `category` instance.
+Unfortunately, because we do not allow morphisms in `Prop`,
+the morphisms `U ⟶ V` are not just proofs `U ≤ V`, but rather
+`ulift (plift (U ≤ V))`.
+-/
+
+/-!
+We now construct as morphisms various inclusions of open sets.
+-/
+-- This is tedious, but necessary because we decided not to allow Prop as morphisms in a category...
 
+/--
+The inclusion `U ⊓ V ⟶ U` as a morphism in the category of open sets.
+-/
+def inf_le_left (U V : opens X) : U ⊓ V ⟶ U :=
+ulift.up (plift.up inf_le_left)
```

Comment:
Since the `ulift.up (plift.up _)` pattern is used a lot, maybe it would be nice to have a constructor for this, `U ≤ V → (U ⟶ V)` for any preorder category.

Reply:
(And its inverse, for that matter)

Reply:
Excellent suggestion! I called these `le_of_hom` and `hom_of_le`.